tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,147 @@
+            <Section
+              horizontal
+              fill
+              height={'95px'}
+              title={'Regeneration and Radius'}>
+              <Section height={'20px'}>
+                <ProgressBar
+                  height={'20px'}
+                  title="Regeneration rate"
+                  value={current_regeneration}
+                  minValue={0}
+                  maxValue={max_regeneration}
+                  ranges={{
+                    'good': [max_regeneration * 0.75, max_regeneration],
+                    'average': [
+                      max_regeneration * 0.25,
+                      max_regeneration * 0.75,
+                    ],
+                    'bad': [0, max_regeneration * 0.25],
+                  }}>
+                  Regeneration {current_regeneration}/{max_regeneration}
+                </ProgressBar>
+              </Section>
+              <Section>
+                <ProgressBar
+                  height={'20px'}
+                  title="Shield radius"
+                  value={current_radius}
+                  minValue={0}
+                  maxValue={max_radius}
+                  ranges={{
+                    'good': [max_radius * 0.75, max_radius],
+                    'average': [max_radius * 0.25, max_radius * 0.75],
+                    'bad': [0, max_radius * 0.25],
+                  }}>
+                  Radius {current_radius}/{max_radius}
+                </ProgressBar>
+              </Section>
+              <Section title={'Settings'} bold={1}>
+                <Section>
+                  <Section>
+                    <NumberInput
+                      title={'Set Radius'}
+                      disabled={active}
+                      value={current_radius}
+                      minValue={3}
+                      maxValue={max_radius}
+                      stepPixelSize={10}
+                      lineHeight="50px"
+                      fontSize="20px"
+                      width="120px"
+                      height="50px"
+                      onChange={(e, value) =>
+                        act('set_radius', {
+                          new_radius: value,
+                        })
+                      }
+                    />
+                    <Button
+                      top={3.25}
+                      right={-16.75}
+                      bold={1}
+                      disabled={recovering}
+                      selected={active}
+                      content={active ? 'On' : 'Off'}
+                      icon="power-off"
+                      onClick={() => act('toggle_shields')}
+                    />
+                    <Button
+                      disabled={active}
+                      right={4.25}
+                      top={3.25}
+                      onClick={() => act('toggle_exterior')}
+                      content={
+                        exterior_only ? 'External only' : 'Internal & External'
+                      }
```

Comment:
`content` is deprecated as the way to write text inside of button, please just put the content as a child, `<Button ...>Text</Button>`